### DISCUSSION TITLE
[Bug]: parseOption.strictMode is false， but parse result always has directive node.

### DISCUSSION THREAD

**Discussion by xusyang** [babel/babel#14789] [Q&A]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

Programmatic API (`babel.transform`, `babel.parse`)

### Input code

```js
import { parse } from '@babel/parser'

const code = `'hello world'`

const ast = parse(code, {
    sourceType: 'module',
    strictMode: false,
    plugins: ['jsx', 'typescript']
 })
```

parseOption.strictMode is false， but parse result always has directive node.

### Current and expected behavior

parseOption.strictMode is false, that not parse Directive Node

### Environment

  System:
    OS: macOS 12.2.1
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.19 - ~/Library/pnpm/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  npmPackages:
    @babel/code-frame: ^7.18.6 => 7.18.6 
    @babel/core: ^7.18.6 => 7.18.6 
    @babel/generator: ^7.18.7 => 7.18.7 
    @babel/parser: ^7.18.6 => 7.18.6 
    @babel/template: ^7.18.6 => 7.18.6 
    @babel/traverse: ^7.18.6 => 7.18.6 
    @babel/types: ^7.18.7 => 7.18.7

**Comment by nicolo-ribaudo** [comment]
The `strictMode` option can only be used to tell Babel to parse the code as strict even if it's parsing a script without an `"use strict"` directive. The logic works like this:
- Are we parsing a module? If yes, we are in strict mode.
- Otherwise, is there a `"use strict"` directive? If yes, we are in strict mode.
- Otherwise, is the `strictMode` option enabled? If yes, we are in strict mode.
- Otherwise, we are not in strict mode.

This has no effect on how directives are parsed: any statement containing only a string at the beginning of a JS program or function is a directive. You can wrap them in parentheses if you want to force them to be expressions:
```js
const code = `('hello world')`
```
or you can prefix them with non-string code:
```js
const code = `0; 'hello world'`
```

Depending on your goal, you may also get the desired using `parseExpression` instead of `parse`.
